CRM/Contribute/Form/Task/Invoice.php

```diff
@@ -543,7 +543,12 @@ public static function printPDF($contribIDs, &$params, $contactIds) {
         self::addActivities($subject, $contribution->contact_id, $fileName, $params);
       }
 
-      CRM_Core_DAO::setFieldValue('CRM_Contribute_DAO_Contribution', $contribution->id, 'invoice_id', $invoiceId);
+      // CRM-20387 save my human-readable invoice number as invoice_id in the contribution table,
+      // but only if the current value is one of: set by this invoicing code, or pay-later, or empty.
+      $set_by_invoicing_code = (0 === strpos($contribution->invoice_id, CRM_Utils_Array::value('invoice_prefix', $prefixValue)));
```

Comment:
If the prefix was changed, this code would overwrite all invoices issued with prior prefix. It's news to me that the value isn't used in generating the invoice or on the user dashboard where they can download invoices, but haven't examined the code... @guanhuan would be able to say better than me. I defer to CompuCorp regarding UK VAT compliance.